### ISSUE TITLE
1.8.0-rc1 still doesn't support Intel 1API

### ISSUE THREAD

**Issue by VictorEijkhout** [libMesh/libmesh#4039]
```
configure: error: Unsupported Intel compiler detected
```

My setup:
```
Using compilers for mode mpi:
 .. CC=mpicc
    where mpicc=/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/bin/mpicc
    and CFLAGS=
 .. CXX=mpicxx
    where mpicxx=/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/bin/mpicxx
    and CXXFLAGS=
 .. FC=ifx
    where ifx=/scratch/projects/compilers/intel24.1/oneapi/compiler/2024.1/bin/ifx
    and FFLAGS=
  where:
    mpicc=/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/bin/mpicc
    show: icx -I"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/include" -L"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -L"/scratch/projects/compilers/intel24.1/oneapi/mp
i/2021.12/lib" -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker "/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -Xlinker -rpath -Xlinker "/scratch/projects/compilers/intel24.1/one
api/mpi/2021.12/lib" -lmpifort -lmpi -lrt -lpthread -Wl,-z,now -Wl,-z,relro -Wl,-z,noexecstack -Xlinker --enable-new-dtags -ldl
    where icx=/scratch/projects/compilers/intel24.1/oneapi/compiler/2024.1/bin/icx
    mpicxx=/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/bin/mpicxx
    show:  icpx -I"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/include" -L"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -L"/scratch/projects/compilers/intel24.1/oneapi/
mpi/2021.12/lib" -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker "/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -Xlinker -rpath -Xlinker "/scratch/projects/compilers/intel24.1/o
neapi/mpi/2021.12/lib" -lmpicxx -lmpifort -lmpi -lrt -lpthread -Wl,-z,now -Wl,-z,relro -Wl,-z,noexecstack -Xlinker --enable-new-dtags -ldl
    where icpx=/scratch/projects/compilers/intel24.1/oneapi/compiler/2024.1/bin/icpx
    mpif90=/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/bin/mpif90
    show:  ifx -I"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/include/mpi/gfortran" -I"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/include" -I"/scratch/projects/compilers/i
ntel24.1/oneapi/mpi/2021.12/include/mpi" -L"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -L"/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -Xlinker --enable-new-dtags
 -Xlinker -rpath -Xlinker "/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -Xlinker -rpath -Xlinker "/scratch/projects/compilers/intel24.1/oneapi/mpi/2021.12/lib" -lmpifort -lmpi -lrt
-lpthread -Wl,-z,now -Wl,-z,relro -Wl,-z,noexecstack -Xlinker --enable-new-dtags -ldl
    where ifx=/scratch/projects/compilers/intel24.1/oneapi/compiler/2024.1/bin/ifx
 .. configuring in /work/00434/eijkhout/libmesh/libmesh-1.8.0-rc1
Configuring as:
./configure --prefix=/work/00434/eijkhout/libmesh/installation-libmesh-1.8.0-rc1-milan-intel24.1-impi21.12 --libdir=/work/00434/eijkhout/libmesh/installation-libmesh-1.8.0-rc1-milan-intel24.1-impi21
.12/lib --enable-silent-rules --enable-unique-id --disable-warnings --with-thread-model=openmp --disable-maintainer-mode --enable-hdf5 --enable-petsc-hypre-required --enable-metaphysicl-required --e
nable-xdr-required --with-cxx-std-min=2014 --without-gdb-command --with-methods=opt
```

**Comment by jwpeterson** [comment]
Not sure if it will fix this issue, but @nmnobre has been looking into configure support for Intel compilers recently, see:  https://github.com/libMesh/autoconf-submodule/pull/17. Once that submodule update is merged into libmesh master, I can also cherry-pick it onto the release branch for the next release candidate.

**Comment by nmnobre** [comment]
Yes, this is definitely fixed. `devel` already has a preliminary fix that'll get you past that configure error, and libmesh/autoconf-submodule#17 further improves support with a slightly better optimisation flags choice.

**Comment by VictorEijkhout** [comment]
Meaning it's fixed post `rc1`? If so I'll wait for the official release.

**Comment by jwpeterson** [comment]
I'll comment here when rc2 is available for testing with this update.

**Comment by jwpeterson** [comment]
@VictorEijkhout would you mind trying the [1.8.0-rc2](https://github.com/libMesh/libmesh/releases/tag/v1.8.0-rc2) release and see if that fixes the issue you reported?

**Comment by VictorEijkhout** [comment]
rc2 seems to build and test correctly with intel 2024.2.  Thanks.

**Comment by jwpeterson** [comment]
OK, thank you for checking!
